src/components/tasks/TaskCommandLogs.tsx

```diff
@@ -11,23 +11,37 @@ import Tooltip from '@mui/material/Tooltip';
 import { WithStyles } from '@mui/styles';
 import createStyles from '@mui/styles/createStyles';
 import withStyles from '@mui/styles/withStyles';
-import GetApp from '@mui/icons-material/GetApp';
+import OpenInNewOutlinedIcon from '@mui/icons-material/OpenInNewOutlined';
+import ArchiveOutlinedIcon from '@mui/icons-material/ArchiveOutlined';
 import Fab from '@mui/material/Fab';
-import { TaskCommandLogsTailQuery } from './__generated__/TaskCommandLogsTailQuery.graphql';
-import { TaskCommandStatus } from './__generated__/TaskCommandList_task.graphql';
+import {
+  TaskCommandLogsTailQuery,
+  TaskCommandLogsTailQueryResponse,
+} from './__generated__/TaskCommandLogsTailQuery.graphql';
+import { TaskCommandStatus, TaskCommandType } from './__generated__/TaskCommandList_task.graphql';
 
 function logURL(taskId: string, command) {
   return 'https://api.cirrus-ci.com/v1/task/' + taskId + '/logs/' + command.name + '.log';
 }
 
+function cacheURL(taskId: string, command, executionInfo) {
+  let cacheKey = executionInfo.cacheRetrievalAttempts.hits.find(hit => hit.key.startsWith(`${command.name}-`)).key;
```

Comment:
`find` here might not actually find anything and then there will be an NPE (null pointer exception). I think it will make sense to move the find logic out of `cacheURL` function. Find a hit in `TaskCommandRealTimeLogs ` and if it's there and `command.type === 'CACHE'` then render the button.